usl_models/tests/atmo_ml/datasets_test.py

```diff
@@ -0,0 +1,102 @@
+import io
+from unittest import mock
+
+from google.cloud import storage  # type: ignore
+import numpy
+
+from usl_models.atmo_ml import datasets
+from usl_models.atmo_ml import cnn_inputs_outputs
+
+
+@mock.patch.object(datasets, "create_atmo_dataset")
+def test_atmo_dataset(mock_create_dataset) -> None:
+    """Create expected dataset for the atmo model from GCS objects."""
+    mock_storage_client = mock.MagicMock(spec=storage.Client)
+
+    # Set parameters for the dataset
+    bucket_name = "our-bucket-name"
+    spatiotemporal_file_names = [
+        "spatiotemporal_data_1.npy",
+        "spatiotemporal_data_2.npy",
+    ]
+    label_file_names = ["labels_1.npy", "labels_2.npy"]
+    spatial_file_name = "spatial_data.npy"
+    lu_index_file_name = "lu_index.npy"
+    time_steps_per_day = 6
+    batch_size = 4
+
+    # Set up mock data for the GCS
+    stinp_1 = numpy.random.rand(batch_size, 100, 100, 1)
+    labelinput_1 = numpy.random.rand(batch_size, 100, 100, 1)
+    stinp_2 = numpy.random.rand(batch_size, 100, 100, 1)
+    labelinput_2 = numpy.random.rand(batch_size, 100, 100, 1)
+
+    # Call the function to process inputs and labels
+    mock_spatiotemporal_data_1, mock_labels_1 = cnn_inputs_outputs.divide_into_days(
+        stinp_1, labelinput_1, input_steps_per_day=4, label_steps_per_day=8
+    )
+    mock_spatiotemporal_data_2, mock_labels_2 = cnn_inputs_outputs.divide_into_days(
+        stinp_2, labelinput_2, input_steps_per_day=4, label_steps_per_day=8
+    )
+    mock_spatial_data = numpy.random.rand(100, 100, 1)
+    mock_lu_index_data = numpy.random.randint(1, 62, (100, 100))
+
+    def mock_blob_func(blob_name):

```

Comment:
we may refactor mock_blob_func to avoid repetition by storing the mock data in a dictionary.
def mock_blob_func(blob_name):
    mock_data_dict = {
        "spatiotemporal_data_1.npy": mock_spatiotemporal_data_1,
        "spatiotemporal_data_2.npy": mock_spatiotemporal_data_2,
        "labels_1.npy": mock_labels_1,
        ....
        "lu_index.npy": mock_lu_index_data,
    }

Reply:
It is also a good option, but i have chosen the elif because it is just in the testing not prod, and we just have few blob names. It directly maps the blob name to mock data, and anyone reading the tests can easily understand what's happening.